docs.source/content/en/docs/How it works/_index.md

```diff
@@ -21,6 +21,201 @@ uint16be Reserved  | 0x0000
 []byte   L3OrJSON  | L3 or JSON
 ```
 
+## JSON messages
+
+JSON messages are used to configure the agent. There are 3 types of messages:
+
+- `request` are sent from the manager to the agent
+- `result` messages are sent from the agent to the manager as a response to a `request`
+- `event` are messages sent from the agent to the manager to indicate an independent event.
+
+Messages always have the following structure:
+
+```json
+{
+  "type": "request|response|event",
+  "body": { ... }
+}
+```
+
+### The `request` body
+
+The request body has the following structure:
+
+```json
+{
+    "id": 1, //Unique ID for this request
+    "op": "operation key", //Currently the only operation supported is "configure"
+    "args": { ... }
+}
+```
+
+### The `configure` message arguments
+
+The `configure` message has the following arguments:
+
+```json
+{
+  "me": "1.2.3.4",
+  "forwards": [
+    // See Forward below
+  ],
+  "others": [
+    // See IPPortProto below
+  ],
+  "HostnameMap": {
+     "hostname": "ip"
+  },
+  "http": {
+    // See HTTP below
+  },
+  "socks": {
+    // See "Socks" below
+  },
+  "loopback": {
+    // See "Loopback" below
+  },
+  "statedir": {
+    "path": "/var/run/norouter...",
+    "disable": false,
+  },
+  "writeEtcHosts": true,
+  "routes": [
+    // See "Route" below
+  ],
+  "nameServers": [
+    // See IPPortProto below
+  ]
+}
+```
+
+### The `Forward` object
+
+The forward object has the following fields:
+
+```json
+{
+  "listen_port": 23
```

Comment:
Who uses 23/tcp (telnet) today? 😛

Reply:
You'd be surprised. :D Anyway, it was meant as a joke. :D

Reply:
Do you want me to change it?

Reply:
Either is fine :+1: